fem/pbilinearform.hpp

```diff
@@ -173,6 +173,30 @@ class ParBilinearForm : public BilinearForm
        vectors on the true dofs. */
    void TrueAddMult(const Vector &x, Vector &y, const real_t a = 1.0) const;
 
+   /// Compute $ y^T M x $
+   /** @warning The calculation is performed on local dofs, assuming that
+       the local vectors are consistent with the prolongations of the true
+       vectors (see ParGridFunction::Distribute()). If this is not the case,
+       use TrueInnerProduct(const ParGridFunction &, const ParGridFunction &)
+       instead.
+       @note It is assumed that the local matrix is assembled and it has
+       not been replaced by the parallel matrix through FormSystemMatrix().
+       @see TrueInnerProduct(const ParGridFunction&, const ParGridFunction&) */
+   real_t InnerProduct(const ParGridFunction &x,
+                       const ParGridFunction &y) const;
+
+   /// Compute $ y^T M x $ on true dofs (grid function version)
+   /** @note It is assumed that the parallel system matrix is assembled,
+       see FormSystemMatrix().
+       @see InnerProduct(const ParGridFunction&, const ParGridFunction&) */
```

Comment:
It is good to clarify in the doxygen that the input `ParGridFunction`s are restricted to true-vectors and then used to compute the inner product.

Reply:
That true dofs in the short description say that, but it can be more verbose 😉 .